### PR TITLE
feature: support REGEXP_REPLACE function

### LINKED ISSUE #3057: REGEXP_REPLACE() | Replace substrings matching regular expression

### PR DESCRIPTION
**Pull request by nange**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* Support REGEXP_REPLACE function
* Make the logic of arguments validating more reasonable and consistent for `regexp_*` functions (the same as snowflake)
* Mini fix for how-to-write-scalar-functions doc

## Changelog

- New Feature
- Improvement
- Documentation

## Related Issues

Fixes #3057

### CODE REVIEW

File: common/functions/src/scalars/strings/regexp_like.rs [hunk 1]
```diff
@@ -100,6 +100,10 @@ impl Function for RegexpLikeFunction {
             mt,
         )?))
     }
+
+    fn passthrough_constant(&self) -> bool {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/regexp_like.rs:104]: I think it's incorrect.

`passthrough_constant` only works when all arguments are constant. So it's ok to `check_get` ConstColumn.

File: common/functions/src/scalars/strings/regexp_instr.rs [hunk 2]
```diff
@@ -69,7 +74,7 @@ impl Function for RegexpInStrFunction {
     }
 
     fn return_type(&self) -> DataTypePtr {
-        u64::to_data_type()
+        NullableType::arc(u64::to_data_type())
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/regexp_instr.rs:77]: Why do we return `NullableType` and `disable_passthrough_null`?

It makes the code more complex.

Comment (sundy-li) [reply common/functions/src/scalars/strings/regexp_instr.rs:77]: Oh, it's same as snowflake did.

File: common/functions/src/scalars/strings/regexp_like.rs (hunk 1 above)
Comment (nange) [reply common/functions/src/scalars/strings/regexp_like.rs]: Thanks, fixed.  And I also have simplified some of the code for regexp_like function.

**Review by sundy-li — APPROVED** [review]

File: common/functions/src/scalars/strings/regexp_instr.rs (hunk 2 above)
Comment (nange) [reply common/functions/src/scalars/strings/regexp_instr.rs:77]: Yes. If we enable the `passthrough_null` , we can't know original of the argument(eg. pos) is `0` or `null` in `eval` function.

**Pull request merged**